packages/sdk/utils/src/account.ts

```diff
@@ -55,70 +56,301 @@ export async function generateMnemonic(
   return bip39ToUse.generateMnemonic(strength, undefined, getWordList(language))
 }
 
-export function validateMnemonic(mnemonic: string, bip39ToUse: Bip39 = bip39Wrapper) {
+export function validateMnemonic(
+  mnemonic: string,
+  bip39ToUse: Bip39 = bip39Wrapper,
+  language?: MnemonicLanguages
+) {
+  if (language !== undefined) {
+    return bip39ToUse.validateMnemonic(mnemonic, getWordList(language))
+  }
+
   const languages = getAllLanguages()
-  for (const language of languages) {
-    if (bip39ToUse.validateMnemonic(mnemonic, getWordList(language))) {
+  for (const guessedLanguage of languages) {
+    if (bip39ToUse.validateMnemonic(mnemonic, getWordList(guessedLanguage))) {
```

Comment:
Or actually I see you have a detectMnemonicLanguage function below. Why not use that here?

Reply:
Its probably slightly faster to use `bip39.validateMnemonic` because that function will bail on the first invalid word instead of comparing the phrase to all languages. Mostly though, this way of doing it feel more direct and makes sense to me personally.